app/scripts/views/settings/communication_preferences.js

```diff
@@ -93,12 +94,23 @@ define(function (require, exports, module) {
     },
 
     setOptInStatus (newsletterId, isOptedIn) {
-      var method = isOptedIn ? 'optIn' : 'optOut';
+      let method, eventPrefix;
+      if (isOptedIn) {
+        method = 'optIn';
+        eventPrefix = '';
+      } else {
+        method = 'optOut';
+        eventPrefix = 'un';
+      }
+
       this.logViewEvent(method);
 
       return this.getMarketingEmailPrefs()[method](newsletterId)
         .then(() => {
           this.logViewEvent(method + '.success');
+          // Emit an additional flow event for consistency with
+          // the call to optIn in the account model
+          this.logFlowEvent(`newsletter.${eventPrefix}subscribed`);
```

Comment:
That makes sense, but it hints that our metrics aren't factored correctly. It seems like a lone event should be triggered, and if the event is done as part of a view, the viewName should be a tag of the event.

Originally, I think the viewName was made part of the event name because DataDog struggled with too many tags but didn't care about the number of distinct event names. I am not sure if this reasoning still holds.